### PR TITLE
My work functionalities

### PR DESCRIPTION
**Pull request by kdrtibor**
adds functionality to add / dismiss work items from my work

### CODE REVIEW

File: OctaneVSPlugin/Common/MyWorkUtils.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            }
+        }
+
+        private async static Task<UserItem> CreateNewUserItem(BaseEntity wrappedBaseEntity)
+        {
+            UserItem newUserItem = new UserItem();
+            newUserItem.SetLongValue("origin", 1L);
+            newUserItem.Id = 5;
+            newUserItem.SetValue("is_new", true);
+            newUserItem.SetValue("reason", null);
```
Comment (atothhpe) [inline OctaneVSPlugin/Common/MyWorkUtils.cs]: fix spacing, make no sense

File: OctaneVSPlugin/Common/MyWorkUtils.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            {
+                throw new Exception("Given param entity is not of type: user_item, type is: " + baseEntity.TypeName);
+            }
+        }
+
+        private async static Task<UserItem> CreateNewUserItem(BaseEntity wrappedBaseEntity)
+        {
+            UserItem newUserItem = new UserItem();
+            newUserItem.SetLongValue("origin", 1L);
+            newUserItem.Id = 5;
```
Comment (atothhpe) [inline OctaneVSPlugin/Common/MyWorkUtils.cs]: 5?

File: OctaneVSPlugin/Common/MyWorkUtils.cs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                return;
+            }
+
+            String id = userItem.GetValue("id").ToString();
+            try
+            {
+                octaneService.RemoveFromMyWork(userItem);
+            }
+            catch (Exception)
+            {
```
Comment (atothhpe) [inline OctaneVSPlugin/Common/MyWorkUtils.cs]: WOT

File: OctaneVSPlugin/Common/MyWorkUtils.cs [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        public static async void RemoveFromMyWork(BaseEntity baseEntity)
+        {
+            OctaneServices octaneService = OctaneServices.GetInstance();
+            UserItem userItem = await octaneService.FindUserItemForEntity(baseEntity);
+            if (userItem == null)
+            {
+                return;
+            }
+
+            String id = userItem.GetValue("id").ToString();
```
Comment (atothhpe) [inline OctaneVSPlugin/Common/MyWorkUtils.cs]: Not used

File: OctaneVSPlugin/MainWindowControl.xaml.cs [hunk 5]
```diff
@@ -135,12 +135,26 @@ private void StopWork(object sender)
...
+        {
+            try
+            {
+                if (SelectedItem?.Entity == null)
+                    return;
+                ToolWindowHelper.RemoveFromMyWork(GetSelectedEntity());
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show("Unable add to my work the selected item.\n\n" + "Failed with message: " + ex.Message, ToolWindowHelper.AppName, MessageBoxButton.OK, MessageBoxImage.Error);
```
Comment (atothhpe) [inline OctaneVSPlugin/MainWindowControl.xaml.cs]: never happens bcuz above

File: OctaneVSPlugin/OctaneServices.cs [hunk 6]
```diff
@@ -53,6 +53,10 @@ public EntityService GetEntityService
         private WorkspaceContext workspaceContext;
         private SharedSpaceContext sharedSpaceContext;
 
+        WorkspaceUser currentUser;
+        private string lastUser;
+        private string lastPassword;
```
Comment (atothhpe) [inline OctaneVSPlugin/OctaneServices.cs]: not needed

File: OctaneVSPlugin/OctaneServices.cs [hunk 7]
```diff
@@ -245,6 +286,24 @@ public async Task<Comment> CreateCommentAsync(Comment entity)
             return createdEntity;
         }
 
+        ///<summary>
+        ///Adds an entity to my work
+        /// </summary>
+        public async void AddToMyWork(UserItem entity)
+        {
+            RestConnector.AwaitContinueOnCapturedContext = false;
+            var craetedEntity = await es.CreateAsync(workspaceContext, entity, null);
```
Comment (atothhpe) [inline OctaneVSPlugin/OctaneServices.cs]: typo

File: OctaneVSPlugin/View/ToolWindowHelper.cs [hunk 8]
```diff
@@ -100,6 +102,46 @@ public static void ViewDetails(BaseEntity entity)
...
+            {
+                if (entity == null)
+                    return;
+
+                MyWorkUtils.AddToMyWork(entity);
+                OctaneMyItemsViewModel.Instance.LoadMyItemsAsync();
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show("Unable to add item to my work.\n\n" + "Failed with message: " + ex.Message, ToolWindowHelper.AppName, MessageBoxButton.OK, MessageBoxImage.Error);
```
Comment (atothhpe) [inline OctaneVSPlugin/View/ToolWindowHelper.cs]: so which catch block are we using

File: OctaneVSPlugin/ViewModel/OctaneMyItemsViewModel.cs [hunk 9]
```diff
@@ -19,6 +19,7 @@
 using MicroFocus.Adm.Octane.VisualStudio.View;
 using Microsoft.VisualStudio.PlatformUI;
 using System;
+using System.Linq;
```
Comment (atothhpe) [inline OctaneVSPlugin/ViewModel/OctaneMyItemsViewModel.cs]: needed?

File: OctaneVSPlugin/ViewModel/OctaneMyItemsViewModel.cs [hunk 10]
```diff
@@ -34,6 +35,10 @@ public class OctaneMyItemsViewModel : INotifyPropertyChanged
         private MainWindowMode _mode;
         private readonly ObservableCollection<OctaneItemViewModel> _myItems;
 
+        private List<MyWorkItemsSublist> myWorkItemSublislts;
+
+        //private readonly IList<> filterItemsList;
```
Comment (atothhpe) [inline OctaneVSPlugin/ViewModel/OctaneMyItemsViewModel.cs]: needed?

File: OctaneVSPlugin/ViewModel/OctaneMyItemsViewModel.cs (hunk 9 above)
Comment (kdrtibor) [reply OctaneVSPlugin/ViewModel/OctaneMyItemsViewModel.cs]: yes, Linq is needed to apply stream operations on collections

**Pull request merged**